nd4j/nd4j-backends/nd4j-tests/src/test/java/org/nd4j/linalg/factory/ops/NDBaseTest.java

```diff
@@ -0,0 +1,923 @@
+        INDArray x = Nd4j.linspace(DataType.DOUBLE, 1.0, 1.0, 5);
+        int numPartitions = 2;
+        int[] partitions = new int[]{1, 0, 0, 1, 0};
+        INDArray y = base.dynamicPartition(x, partitions, numPartitions);
+        //TODO: crashes here. Op needs fixing.
+
+    }
+
+    @Test
+    public void testDynamicStitch() {
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.linspace(DataType.DOUBLE, 1.0, 1.0, 9).reshape(3, 3);
+        INDArray y = base.dynamicStitch(new INDArray[]{x, x}, 0);
+        //TODO: crashes here. Op needs fixing.
+    }
+
+    @Test
+    public void testScalarEq() {
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.zeros(DataType.DOUBLE, 3, 3);
+        INDArray y = base.eq(x, 0.0);
+        INDArray y_exp = Nd4j.createFromArray(new boolean[][]{{true, true, true}, {true, true, true}, {true, true, true}});
+        assertEquals(y, y_exp);
+    }
+
+    @Test
+    public void testEq() {
+        //element wise  eq.
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.zeros(DataType.DOUBLE, 3, 3);
+        INDArray y = base.eq(x, x);
+        INDArray y_exp = Nd4j.createFromArray(new boolean[][]{{true, true, true}, {true, true, true}, {true, true, true}});
+        assertEquals(y, y_exp);
+    }
+
+    @Test
+    public void testExpandDims() {
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.zeros(DataType.DOUBLE, 1, 2);
+        INDArray y = base.expandDims(x, 0);
+        INDArray y_exp = Nd4j.createFromArray(new double[][]{{1.0, 0.0, 0.0}});
+        //TODO: Fix. Not getting the expected output.
+        assertTrue(y.equalsWithEps(y_exp, 1e-6));
+    }
+
+    @Test
+    public void testFill() {
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.createFromArray(2, 2);
+        INDArray y = base.fill(x, DataType.DOUBLE, 1.1);
+        INDArray y_exp = Nd4j.createFromArray(new double[][]{{1.1, 1.1}, {1.1, 1.1}});
+        // TODO: Fails: y is a float, not a double as expected.
+        assertTrue(y.equalsWithEps(y_exp, 1e-1));
+    }
+
+    @Test
+    public void testGather() {
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.zeros(DataType.DOUBLE, 3, 3);
+        int[] ind = new int[]{0};
+        INDArray y = base.gather(x, ind, 0);
+        //TODO: crashes here. Op needs fixing.
+    }
+
+    @Test
+    public void testGatherNd() {
+        NDBase base = new NDBase();
+
+        assertTrue(false); // TODO: Fix the Op. (current signature is wrong.)
+    }
+
+    @Test
+    public void testScalarGt() {
+        //Scalar gt.
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.zeros(DataType.DOUBLE, 3, 3);
+        INDArray y = base.gt(x, -0.1);
+        INDArray y_exp = Nd4j.createFromArray(new boolean[][]{{true, true, true}, {true, true, true}, {true, true, true}});
+        assertEquals(y, y_exp);
+        //TODO: Passes, but I do not like the detailed memory management in the Op constructor.
```

Comment:
Already flagged, should be removed in constructor.

Reply:
The +1 here just means I pasted this comment into the test. Will need to resolve the constructor.